google/auth/impersonated_credentials.py

```diff
@@ -321,6 +321,47 @@ def with_quota_project(self, quota_project_id):
             iam_endpoint_override=self._iam_endpoint_override,
         )
 
+    @classmethod
+    def from_authorized_user_info(cls, source_credentials, info, scopes=None):
```

Comment:
Thank you @MadJlzz for working on it! 

The high level idea of this PR looks good. Here is my thoughts on it:

1. Can we rename this function to "from_impersonated_credentials_info"? authorized_user has a special meaning which is for user account.
2. Why not extend this function to handle source_credentials deserialization, so that you don't need to create the source credentials in the _default? 
3. When deserializing the source credential, can we firstly check its type? It currently assumes the source is user account credentials, but it is not necessarily true. We should at least support user account credentials and service account credentials.